tools/archinfo.js

```diff
@@ -125,48 +125,52 @@ var os = require('os');
 // a fiber. Throws an error if it's not a supported architecture.
 //
 // If you change this, also change scripts/admin/launch-meteor
-var _host = null; // memoize
-var host = function () {
-  if (! _host) {
-    var run = function (...args) {
-      var result = files.run.apply(null, args);
-      if (! result)
-        throw new Error("can't get arch with " + args.join(" ") + "?");
-      return result.replace(/\s*$/, ''); // trailing whitespace
-    };
 
-    var platform = os.platform();
+let _host = null; // memoize
+function getHost() {
+  if (!_host) {
+    function run(...args) {
+      const result = files.run.apply(null, args);
 
-    if (platform === "darwin") {
-      // Can't just test uname -m = x86_64, because Snow Leopard can
-      // return other values.
-      if (run('uname', '-p') !== "i386" ||
-          run('sysctl', '-n', 'hw.cpu64bit_capable') !== "1")
-        throw new Error("Only 64-bit Intel processors are supported on OS X");
-      _host  = "os.osx.x86_64";
-    }
+      if (!result) {
+        throw new Error('can\'t get arch with ' + args.join(' ') + '?');
```

Comment:
Yeah I'm not really sure what the deal is with apostrophes... but I'm pretty sure single quotes are the standard these days. Personally I would be OK with using backticks to avoid the escaping.